sql/sys_vars.cc

```diff
@@ -7173,3 +7173,23 @@ static Sys_var_optimizer_cost Sys_optimizer_scan_cost(
   CMD_LINE(REQUIRED_ARG),
   VALID_RANGE(0, 100000000), DEFAULT(DEFAULT_TABLE_SCAN_SETUP_COST),
   COST_ADJUST(1000));
+
+static Sys_var_charptr_fscs Sys_server_redirect_target(
+	"server_redirect_target",
+        "Server redirection target. This should be a hostname, an IP address, or "
+        "a hostname or IP address followed by ':PORT'. Server redirection will "
+        "not be activated unless SERVER_REDIRECT_MODE is also set.",
+        GLOBAL_VAR(server_redirect_target), CMD_LINE(REQUIRED_ARG),
+        DEFAULT(NULL));
+
+static const char *server_redirect_mode_names[]= {
+        "OFF", "ON", "ALL", 0 };
+
+static Sys_var_enum Sys_server_redirect_mode(
+       "server_redirect_mode",
+       "Server redirection mode. "
+       "Possible modes are: OFF - No redirection, "
+       "ON: Redirect all new clients attempting to connect over the network to the standard server port (no redirection of local socket-based connections, nor of connections to the EXTRA_PORT), "
+       "ALL: Unconditionally redirect new clients attempting to connect (even via local socket-based connections).",
+       GLOBAL_VAR(server_redirect_mode), CMD_LINE(REQUIRED_ARG),
+       server_redirect_mode_names, DEFAULT(0));
```

Comment:
Over at https://jira.mariadb.org/browse/MDEV-15935?focusedCommentId=263403#comment-263403, @markus456 suggested renaming the `ON` value to `FORCE`.

> It would cover use-cases other than forceful redirection and also make it possible to have a longer redirection period. As such, I'd name the value for `server_redirect_mode` as `FORCE` rather than `ON`: this way the door is left open for a less intrusive method of redirection

That's a good idea :+1:. It's clearer and leaves open more options for future "less-forceful" redirection modes.